Adds CFF

sapetti9 opened the pull request:
Adds citation file format in order to make it easier to cite the Frictionless Python Framework in academic publications. @peterdesmet please have a look and see what you think.

- fixes #<1693>

cc @pierrecamilleri & @roll -- I added your email too, but please remove it if you don't wish for it to be here.

sapetti9 committed:
Adds CFF

Adds citation file format in order to make it easier to cite the Frictionless Python Framework in academic publications

sapetti9 committed:
Update CITATION.cff

sapetti9 committed:
Update CITATION.cff

peterdesmet commented on CITATION.cff:
```diff
@@ -0,0 +1,81 @@
+# Visit https://bit.ly/cffinit to generate yours today!
+
+cff-version: 1.2.0
+title: Frictionless Python Framework
```
@roll the package is installed as `frictionless`, the repo/README title is called `frictionless-py`, the website is https://framework.frictionlessdata.io/

I suggest to call it:

```suggestion
title: 'frictionless: Python library for Data Packages'
```

peterdesmet commented on CITATION.cff:
```diff
@@ -0,0 +1,81 @@
+authors:
+  - given-names: Evgeny
+    family-names: Karev
+    email: eskarev@gmail.com
```
Affiliation `Open Knowledge Foundation`?

sapetti9 replied:
@roll do you want to be listed as affiliated to OKFN or you'd rather not?

roll replied:
@peterdesmet 
Does it mean historical affiliation?

It's historical then OKFN if current Datist

peterdesmet replied:
You'd say current one. You can update it when it changes or leave blank. Since you're the first author, I would indicate it though.

peterdesmet commented on CITATION.cff:
```diff
@@ -0,0 +1,81 @@
+  - given-names: Peter
+    family-names: Desmet
+    orcid: 'https://orcid.org/0000-0002-8442-8025'
+    affiliation: INBO
```
```suggestion
    affiliation: Research Institute for Nature and Forest (INBO)
```

peterdesmet commented on CITATION.cff:
```diff
@@ -0,0 +1,81 @@
+    email: sara.petti@okfn.org
+identifiers:
+  - type: doi
+    value: 10.5281/zenodo.4663759
```
This is the DOI for version 5.18 specifically. It is fine to use that one if you're going to update this every time a new version is created (cf. just like you have to update the `version` and `date-released`). If don't want to do that, refer to the versionless DOI https://doi.org/10.5281/zenodo.4663759.

Note: I would refer to versionless DOI in README.

peterdesmet commented on CITATION.cff:
```diff
@@ -0,0 +1,81 @@
+identifiers:
+  - type: doi
+    value: 10.5281/zenodo.4663759
+repository-code: 'https://github.com/frictionlessdata/frictionless-py'
```
I suggest to also include the pypi link, as `repository` (is what we do for CRAN in R) or `repository-artifact`: https://github.com/citation-file-format/citation-file-format/blob/1.2.0/schema-guide.md#repository-artifact

```suggestion
repository: 'https://pypi.org/project/frictionless/'
repository-code: 'https://github.com/frictionlessdata/frictionless-py'
```

peterdesmet commented on CITATION.cff:
```diff
@@ -0,0 +1,81 @@
+  powered by the lightweight yet comprehensive Frictionless
+  Data Package (https://datapackage.org/).
+license: MIT
+commit: >-
```
I would not include this, unless you want to maintain it. Referring to version is sufficient imo.

peterdesmet commented on CITATION.cff:
```diff
@@ -0,0 +1,81 @@
+    value: 10.5281/zenodo.4663759
+repository-code: 'https://github.com/frictionlessdata/frictionless-py'
+url: 'https://framework.frictionlessdata.io/'
+abstract: >-
```
It is good that this abstract is reused in the README and repo description, but I think it could benefit from an update. That's likely a separate issue though.

sapetti9 committed:
Apply suggestions from code review

Thanks @peterdesmet, this looks much better now. I've left @roll's affiliation as an open question to him, and lets start a separate discussion about the description.

Co-authored-by: Peter Desmet <peter.desmet@inbo.be>

sapetti9 committed:
Update CITATION.cff

roll commented:
Great work @sapetti9 !

sapetti9 committed:
Update CITATION.cff

Updating Evgeny's affiliation

pierrecamilleri commented:
Thanks ! 

Not sure what is happening with the CI though, I'm looking into it.

pierrecamilleri commented:
Still not clear what exactly happened but it seems to be a regression linked to sqlalchemy and/or duckdb (see for instance [here](https://github.com/sqlalchemy/sqlalchemy/discussions/12011), ~~but I could not find a relevant associated duckdb issue)~~ see edit. 

I try to fix this quickly by pinpointing the versions.

EDIT : I think I found [the issue](https://github.com/Mause/duckdb_engine/issues/1128)

pierrecamilleri commented:
Issue solved, thanks for your patience !

peterdesmet commented:
Great! @sapetti9 or @pierrecamilleri can you merge this?

sapetti9 commented:
Done. Thanks all!
